app/controllers/main/routes.py

```diff
@@ -45,42 +46,99 @@ def events(selectedTerm=None):
-    if not g.current_user.isCeltsAdmin and not g.current_user.isCeltsStudentStaff and g.current_user.username != username:
-        return "Access Denied", 403
+@main_bp.route('/profile/<username>', methods=['GET'])
+def viewVolunteersProfile(username):
+    """
+    This function displays the information of a volunteer to the user
+    """
     try:
-        profileUser = User.get(User.username == username)
-        upcomingEvents = getUpcomingEventsForUser(username)
-        programs = Program.select()
-        interests = Interest.select().where(Interest.user == profileUser)
-        interests_ids = [interest.program.id for interest in interests]
-        rsvpedEventsList = EventRsvp.select().where(EventRsvp.user == profileUser)
-        rsvpedEvents = [event.event.id for event in rsvpedEventsList]
-
-        return render_template('/volunteer/volunteerProfile.html',
-                               title="Volunteer Interest",
-                               user = profileUser,
-                               programs = programs,
-                               interests = interests,
-                               interests_ids = interests_ids,
-                               upcomingEvents = upcomingEvents,
-                               rsvpedEvents = rsvpedEvents)
+        User.get(User.username == username)
+    except Exception as e:
+        return "User does not exist", 404
+    if (g.current_user.username == username) or g.current_user.isAdmin:
+         upcomingEvents = getUpcomingEventsForUser(username)
+         programs = Program.select()
+         interests = Interest.select().where(Interest.user == username)
+         programsInterested = [interest.program for interest in interests]
+         eligibilityTable = []
+         for program in programs:
+              notes = ProgramBan.select().where(ProgramBan.user == username,
+                                                ProgramBan.program == program,
+                                                ProgramBan.endDate > datetime.datetime.now())
+              noteForDict = "None"
+              for j in notes:
+                  noteForDict = j.banNote.noteContent
+              eligibilityTable.append({"program" : program,
+                                       "completedTraining" : (username in trainedParticipants(program)),
+                                       "isNotBanned" : isEligibleForProgram(program, username),
+                                       "banNote": noteForDict})
+         return render_template ("/main/volunteerProfile.html",
+            programs = programs,
+            interests = interests,
+            programsInterested = programsInterested,
+            upcomingEvents = upcomingEvents,
+            eligibilityTable = eligibilityTable,
+            volunteer = User.get(User.username == username),
+            user = g.current_user)
+    abort(403)
+
+
+@main_bp.route('/banUser/<program_id>/<username>', methods=['POST'])
```

Comment:
I think these urls should match the interest url format, like `/<username>/ban/<program_id>` and `/<username>/unban/<program_id>`